Youtube-Ad-blocker-Reminder-Remover.user.js

```diff
@@ -225,10 +225,11 @@
         }
     
         videoPlayerElements.forEach(videoPlayerElement => {
-            while (videoPlayerElement.firstChild) {
-                videoPlayerElement.removeChild(videoPlayerElement.firstChild);
-            }
+        const iframes = videoPlayerElement.querySelectorAll('iframe');
+        iframes.forEach(iframe => {
+            iframe.remove();
         });
+    });

```

Comment:
Why not leave the previous while loop and then make a new forEach to also remove iframes

Reply:
like this?
function clearAllPlayers() {
    
        const videoPlayerElements = document.querySelectorAll('.html5-video-player');
    
        if (videoPlayerElements.length === 0) {
            console.error("No elements with class 'html5-video-player' found.");
            return false;
        }
    
        videoPlayerElements.forEach(videoPlayerElement => {
        while (videoPlayerElement.firstChild) {
            videoPlayerElement.removeChild(videoPlayerElement.firstChild);
        }
        const iframes = videoPlayerElement.querySelectorAll('iframe');
        iframes.forEach(iframe => {
            iframe.remove();
        });
    });
    
        console.log("Removed all current players!");
        return true;
    }

Reply:
Yup that ensures previous need for the while loop code is maintained while adding onto it the new foreach iframe remove, it may be redundant but test it, that may be better! Thanks